cursorless-talon/src/modifiers/relative_scope.py

```diff
@@ -5,13 +5,77 @@
 mod = Module()
 
 
-@mod.capture(rule="(previous | next) <user.cursorless_scope_type>")
+@mod.capture(rule="previous | next")
+def cursorless_relative_direction(m) -> str:
+    """Previous/next"""
+    return "backward" if m[0] == "previous" else "forward"
+
+
+@mod.capture(rule="backward")
+def cursorless_relative_backward() -> str:
+    """Backward"""
+    return "backward"
+
+
+@mod.capture(
+    rule="[<user.ordinals_small>] <user.cursorless_relative_direction> <user.cursorless_scope_type>"
+)
+def cursorless_relative_scope_singular(m) -> dict[str, Any]:
+    """Relative previous/next singular scope. `next funk`"""
+    return create_relative_scope_modifier(
+        m.cursorless_scope_type,
+        getattr(m, "ordinals_small", 1),
+        1,
+        m.cursorless_relative_direction,
+    )
+
+
+@mod.capture(
+    rule="<user.cursorless_relative_direction> <number_small> <user.cursorless_scope_type_plural>"
```

Comment:
Do we want to support a rule that combines this with the above, so that you can say eg `"third next two funks"`?  Our api supports that and I don't think there's any way to express it

Reply:
I'm leaning against it. First I think the grammar is becoming way to verbose to be able to be remembered easily and you can always just use a hat. eg `"two funks air"`. And since we are talking about something that is close to the selection we will probably always have a good hat.

Reply:
Hmm yeah I'm slightly on the fence as well.  Tbh the easiest way to approximate this behaviour is to just chain: `"two funks third next funk"`, which will work with multiple cursors, unlike using a mark

The main reason I'm leaning towards this direction is that I kind of like the idea of orthogonalising position and quantity.  Then we have these ways of specifying position:

- `"first"` / `"second"` / `"last"`
- `"next"` / `"previous"` / `"second next"` / `"second previous"`

And we have these ways of specifying quantity:

- `"funk"`
- `"two funks"`

And we can combine any position with any quantity

But I'm not sure practically speaking how much we'd use this expressivity 🤷‍♂️